File: ext/lynx/stdlib/collections/collection.zep.c

#ifdef HAVE_CONFIG_H
#include "../../../ext_config.h"
#endif

#include <php.h>
#include "../../../php_ext.h"
#include "../../../ext.h"

#include <Zend/zend_operators.h>
#include <Zend/zend_exceptions.h>
#include <Zend/zend_interfaces.h>

#include "kernel/main.h"


/**
 * @author: Patsura Dmitry <zaets28rus@gmail.com>
 */
ZEPHIR_INIT_CLASS(Lynx_Stdlib_Collections_Collection) {

	ZEPHIR_REGISTER_CLASS(Lynx\\Stdlib\\Collections, Collection, lynx, stdlib_collections_collection, lynx_stdlib_collections_collection_method_entry, 0);

	zend_class_implements(lynx_stdlib_collections_collection_ce TSRMLS_CC, 1, spl_ce_Countable);
	zend_class_implements(lynx_stdlib_collections_collection_ce TSRMLS_CC, 1, zend_ce_aggregate);
	return SUCCESS;

}

/**
 * Get elements count from a collection
 */
PHP_METHOD(Lynx_Stdlib_Collections_Collection, count) {

}

/**
 * Get iterator
 */
PHP_METHOD(Lynx_Stdlib_Collections_Collection, getIterator) {

}

/**
 * Execute a function on each element in the collection
 */
PHP_METHOD(Lynx_Stdlib_Collections_Collection, map) {

}

/**
 * Execute a function on each element in the collection and
 * removes the element from the collection if the executed function returns false
 */
PHP_METHOD(Lynx_Stdlib_Collections_Collection, filter) {

}

/**
 * Set the current element to the first element and returns it
 */
PHP_METHOD(Lynx_Stdlib_Collections_Collection, first) {

}

/**
 * Set current element to the last element and returns it
 */
PHP_METHOD(Lynx_Stdlib_Collections_Collection, last) {

}

/**
 * Add a element to the collection
 */
PHP_METHOD(Lynx_Stdlib_Collections_Collection, add) {

}

File: ide/0.0.4/Lynx/stdlib/collections/Collection.zep.php
<?php

namespace Lynx\Stdlib\Collections;

class Collection implements \Countable, \IteratorAggregate
{

    /**
     * Get elements count from collection
     * 	
     */
	abstract public function count();

    /**
     * Get iterator
     * 	
     */
	abstract public function getIterator();

    /**
     * Execute fuction to each element in the collection
     * 	
     *
     * @param \Closure $func 
     */
	abstract public function map($func);

    /**
     * Execute fuction to each element in the collection and
     * remove element from collection if executed function return false
     * 	
     *
     * @param mixed $func 
     */
	abstract public function filter($func);

    /**
     * Set current element to the first element and return it
     * 	
     */
	abstract public function first();

    /**
     * Set current element to the last element and return it
     * 	
     */
	abstract public function last();

    /**
     * Add element to collection
     * 	
     *
     * @param mixed $element 
     */
	abstract public function add($element);

}
